pipeline-runner/R/seurat-2-load_seurat.R

```diff
@@ -132,6 +132,23 @@ reconstruct_seurat <- function(dataset_fpath) {
     names(user_scdata@reductions) <- tolower(names(user_scdata@reductions))
     red_name <- SeuratObject::DefaultDimReduc(user_scdata)
     check_type_is_safe(red_name)
+    red_match <- grep("umap|tsne", red_name, value = TRUE)
+
+    is_umap <- grepl("umap", red_match)
+    is_tsne <- grepl("tsne", red_match)
+
+    if (length(red_match) > 0 && !(red_match %in% c("umap", "tsne"))) {
+      new_red_name <- ifelse(is_umap, "umap", ifelse(is_tsne, "tsne", NA))
+      if (is.na(new_red_name)) {
+        stop("Error: no umap or tsne reductions found.")
```

Comment:
Also you should test the error is thrown in case embeddings are not called umap/tsne

Reply:
the user will not get this specific error, but I think somehow we do propagate pipeline errors. If there are errors in the dimensionality reduction part, the user gets the error " We couldn't get the `pca` and `umap` or `tsne` dimensionality reduction from the uploaded Seurat object. Please check requirements and re-upload after fixing."
![Screen Shot 2023-11-20 at 3 27 10 PM](https://github.com/hms-dbmi-cellenics/pipeline/assets/37797231/023267f2-9425-4146-845e-7892982b4f88)

Reply:
> Also you should test the error is thrown in case embeddings are not called umap/tsne

I actually removed that error. It doesn't make sense given that it's inside a condition that actually implies that there is a match for "umap" or "tsne". So `new_red_name` is never `NA`. And `stopifnot(red_name %in% c('umap', 'tsne'))` [here](https://github.com/hms-dbmi-cellenics/pipeline/blob/eab964bed35d1bbc023f67e1f808619334e0520a/pipeline-runner/R/seurat-2-load_seurat.R#L148), already handles the case where the reduction is neither `umap` nor `tsne`.